src/inetstack/protocols/layer4/mod.rs

```diff
@@ -338,19 +338,39 @@ impl Peer {
     }
 
     /// Pushes a buffer to a TCP socket.
-    pub async fn push(&mut self, sd: &mut Socket, buf: &mut DemiBuffer, addr: Option<SocketAddr>) -> Result<(), Fail> {
+    pub async fn push(
+        &mut self,
+        sd: &mut Socket,
+        bufs: ArrayVec<DemiBuffer, MAX_BATCH_SIZE_NUM_PACKETS>,
+        addr: Option<SocketAddr>,
+    ) -> Result<(), Fail> {
         match sd {
-            Socket::Tcp(socket) => self.tcp.push(socket, buf).await,
-            Socket::Udp(socket) => self.udp.push(socket, buf, addr).await,
+            Socket::Tcp(socket) => self.tcp.push(socket, bufs).await,
+            Socket::Udp(socket) => {
+                for buf in bufs {
+                    trace!("pushing");
```

Comment:
If we need a trace here, then we should add slightly more information in this log statement.